tests/foreman/installer/test_installer.py

```diff
@@ -177,20 +182,60 @@ def sat_non_default_install(module_sat_ready_rhels):
         'enable-foreman-plugin-discovery',
         'foreman-proxy-plugin-discovery-install-images true',
     ]
-    install_satellite(module_sat_ready_rhels[1], installer_args, enable_fapolicyd=True)
-    sat = module_sat_ready_rhels[1]
-    sat.enable_ipv6_http_proxy()
-    sat.execute('dnf -y --disableplugin=foreman-protector install foreman-discovery-image')
-    return sat
+
+    def deploy(request=None, delay=300):
+        rhel_version = get_sat_rhel_version()
+        deploy_args = {
+            'deploy_rhel_version': rhel_version.base_version,
+            'deploy_network_type': 'ipv6' if settings.server.is_ipv6 else 'ipv4',
+            'deploy_flavor': settings.flavors.default,
+            'workflow': settings.server.deploy_workflows.os,
+        }
+        if hasattr(request, 'param'):
+            if isinstance(request.param, dict):
+                deploy_args.update(request.param)
+            else:
+                deploy_args['deploy_rhel_version'] = request.param
+        vmb = Broker(**deploy_args, host_class=Satellite)
+        timeout = (1200 + delay) * 3
+        sat_host = wait_for(vmb.checkout, timeout=timeout, delay=delay, fail_condition=[])
+        sat_host = sat_host.out
+        install_satellite(sat_host, installer_args, enable_fapolicyd=True)
+        sat_host.enable_ipv6_http_proxy()
+        sat_host.execute('dnf -y --disableplugin=foreman-protector install foreman-discovery-image')
+        return sat_host
+
+    return deploy
+
+
+@contextmanager
+def _target_sat_for_capsule(request, module_target_sat, sat_non_default_install):
+    """This is the actual working part of the following target_sat fixtures"""
+    if 'sanity' in request.config.option.markexpr:
+        yield module_target_sat
+    else:
+        new_sat = sat_non_default_install(request)
+        yield new_sat
+        new_sat.teardown()
+        Broker(hosts=[new_sat]).checkin()
+
+
+@pytest.fixture(scope='module')
+def satellite_for_capsule_installer(request, module_target_sat, sat_non_default_install):
```

Comment:
Is name `satellite_for_capsule_installer` appropriate ?

Reply:
What would you suggest? :)

Reply:
This fixture no more exisits.